File: ctfd_challenges/cau2/stego_tool.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
============================================================================
  Challenge 2 - Steganography Tool  (FILL IN THE BLANKS)
============================================================================

BACKGROUND: The file worldcup.png looks like an ordinary image, but the flag is
            hidden as plain TEXT inside the file itself (in the picture's
            metadata). The pixels look completely normal -- yet the flag is
            sitting right there in the bytes of the file.

            You do NOT need to touch the pixels. Just read the raw bytes of
            the file and look for the flag text  UIT{...} .

            Shortcut (no code): the classic `strings` command shows all the
            readable text in a file:
                strings worldcup.png                  (Linux / macOS)
                strings worldcup.png | findstr UIT     (Windows)

USAGE:
    python stego_tool.py worldcup.png

----------------------------------------------------------------------------
 YOUR TASK: implement the 2 blanks marked  >>> FILL <<< :
     (1) read_file_bytes() - read ALL the bytes of the image file
     (2) find_flag()       - locate the  UIT{...}  text inside those bytes
============================================================================
"""
import sys
import re


# ============================================================================
#  (1) >>> FILL <<<
#  Open the file in BINARY mode ("rb") and return ALL of its bytes.
# ============================================================================
def read_file_bytes(image_path):

    return data


# ============================================================================
#  (2) >>> FILL <<<
#  'data' is the raw bytes of the file. Find the flag  UIT{...}  hidden inside
#  it and return it as a string.
#  (Hint: turn the bytes into text, then search for the  UIT{...}  pattern.)
# ============================================================================
def find_flag(data):

    return flag


def main():
    image_path = sys.argv[1] if len(sys.argv) > 1 else "worldcup.png"
    print("=" * 56)
    print(f"  STEGO DECODER -- file: {image_path}")
    print("=" * 56)
    data = read_file_bytes(image_path)
    flag = find_flag(data)
    if flag and flag.startswith("UIT{"):
        print(f"  >>> FLAG FOUND: {flag}")
    else:
        print("  Flag not found. Check your two functions above.")
        print(f"  (Current result: {flag!r})")
    print("=" * 56)


if __name__ == "__main__":
    main()

File: ctfd_challenges/cau2/stego_tool.cpp
// ============================================================================
//   Challenge 2 - Steganography Tool (C++)  --  FILL IN THE BLANKS
// ============================================================================
//  The file worldcup.png looks like an ordinary image, but the flag is hidden
//  as plain TEXT inside the file itself (in its metadata). You do NOT need to
//  touch the pixels -- just read ALL the bytes of the file and look for the
//  flag text  UIT{...} .
//
//  Build:  g++ -std=c++17 -O2 -o stego_tool stego_tool.cpp
//  Run:    ./stego_tool worldcup.png        (Windows: stego_tool.exe worldcup.png)
//
//  YOUR TASK: implement the 2 functions marked  >>> FILL <<< :
//    (1) read_file_bytes() - read ALL the bytes of the image file
//    (2) find_flag()       - locate the  UIT{...}  text inside those bytes
// ============================================================================
#include <iostream>
#include <fstream>
#include <string>
#include <regex>
#include <iterator>

// ============================================================================
//  (1) >>> FILL <<<
//  Open the file in BINARY mode and return ALL of its bytes as a std::string.
// ============================================================================
std::string read_file_bytes(const std::string& path) {

    return "";   // <-- replace with the bytes read from the file
}

// ============================================================================
//  (2) >>> FILL <<<
//  'data' holds all the bytes of the file. Find the hidden UIT{...} string and
//  return it. 
// ============================================================================
std::string find_flag(const std::string& data) {

    return "";   // <-- replace with the flag you found
}

int main(int argc, char** argv) {
    std::string path = (argc > 1) ? argv[1] : "worldcup.png";
    std::cout << "========================================================\n";
    std::cout << "  STEGO DECODER (C++) -- file: " << path << "\n";
    std::cout << "========================================================\n";

    std::string data = read_file_bytes(path);
    std::string flag = find_flag(data);

    if (!flag.empty() && flag.rfind("UIT{", 0) == 0)
        std::cout << "  >>> FLAG FOUND: " << flag << "\n";
    else
        std::cout << "  Flag not found. Complete the two functions above.\n";

    std::cout << "========================================================\n";
    return 0;
}
